src/probnum/diffeq/odefiltsmooth/ivpfiltsmooth.py

```diff
@@ -17,28 +24,172 @@ class GaussianIVPFilter(ODESolver):
 
     Parameters
     ----------
-    gaussfilt : gaussianfilter.GaussianFilter
-        e.g. the return value of ivp_to_ukf(), ivp_to_ekf1().
-
-    Notes
-    -----
-    - gaussfilt.dynamicmodel contains the prior,
-    - gaussfilt.measurementmodel contains the information about the ODE right hand side function,
-    - gaussfilt.initialdistribution contains the information about the initial values.
+    ivp
+        Initial value problem to be solved.
+    prior
+        Prior distribution.
+    measurement_model
+        ODE measurement model.
+    with_smoothing
+        To smooth after the solve or not to smooth after the solve.
+    init_implementation
+        Initialization algorithm. Either via Scipy (``initialize_odefilter_with_rk``) or via Taylor-mode AD (``initialize_odefilter_with_taylormode``).
+        For more convenient construction, consider :func:`GaussianIVPFilter.from_rk_init`.
     """
 
-    def __init__(self, ivp, gaussfilt, with_smoothing):
-        if not isinstance(gaussfilt.dynamics_model, pnfs.statespace.Integrator):
+    def __init__(
+        self,
+        ivp: IVP,
+        prior: pnfs.statespace.Integrator,
+        measurement_model: pnfs.statespace.DiscreteGaussian,
```

Comment:
No more Kalman initialisation.

This came about, because (according to our discussion) I removed the need to compute an initrv before initialising the filter, which made it awkward to initialise the ODE solver with a Kalman object. Since this is more in the direction of where I want to go with the solver API, I made the change now.